ocean/src/clans.rs

```diff
@@ -1,39 +1,58 @@
 
+use std::collections::HashMap;
+
 #[derive(Debug)]
 pub struct ClanSystem {
-    // TODO: add necessary fields
+    pub clans: HashMap<String, Vec<String>>,
```

Comment:
We could also choose to define another class for each Clan for clarity. However, that does have the tradeoff of brevity, so both options seem reasonable.